Reinstate ``astropydev`` tests

tbowers7 opened the pull request:
As titled.

Reverts pypeit/PypeIt#2018

**_Do not merge_** until https://github.com/astropy/astropy/issues/18735 is satisfactorily resolved (as evidenced by the tests listed below all passing on their own).

tbowers7 committed:
Revert "Temporarily skip ``astropydev`` tests"

tbowers7 commented:
Potential fix in https://github.com/astropy/astropy/pull/18753 -- keep an eye on it.

pllim commented:
If we don't get it merged soon enough upstream, feel free to ping me. We apologize for any inconvenience caused!

pllim commented:
I just merged https://github.com/astropy/astropy/pull/18753 . Please try again. Thanks!

tbowers7 committed:
Merge branch 'develop' into revert-2018-tmp_rm_astropydev

pllim commented:
BTW you could pull astropy nightly wheel instead of building from source here:

https://github.com/pypeit/PypeIt/blob/c4a495cc98ea63f2c199c9abea465bddd58ce5d7/tox.ini#L48

But if you do that, I am rebuilding the wheel right now with the patch, so do wait another hour. Thanks!

pllim commented:
ref: https://docs.astropy.org/en/latest/install.html#install-pre-built-development-version

tbowers7 committed:
Test ``astropydev`` against nightly wheel

Run the ``astropydev`` tests against the nightly wheel rather than
building from source each time.

	modified:   tox.ini

tbowers7 commented:
> I just merged [astropy/astropy#18753](https://github.com/astropy/astropy/pull/18753) . Please try again. Thanks!

Thanks @pllim!  Our CI tests now pass against the AstroPy `main` branch.

kbwestfall reviewed (approved):
Looks great.  Thanks!
